apps/sms/test/unit/navigation_test.js

```diff
@@ -521,6 +521,66 @@ suite('navigation >', function() {
     });
   });
 
+  suite('ensureCurrentPanel >', function() {
+    setup(function(done) {
+      this.sinon.stub(Navigation, 'slide').returns(Promise.resolve());
+      this.sinon.spy(Navigation, 'toPanel');
+
+      // Set custom hash to have non-initialized current panel
+      window.location.hash = '#x-unknown-panel';
+
+      Navigation.init().catch((e) => {
+        Navigation.toPanel.reset();
+
+        // It's expected error for the unknown panel
+        if (e.message === 'Panel x-unknown-panel is unknown.') {
+          return;
+        }
+
+        return Promise.reject(e);
+      }).then(done, done);
+    });
+
+    test('navigates to default panel when current panel is not set',
+    function(done) {
+      // We don't have any correct current panel, so navigation to default
+      // panel should be triggered.
+      Navigation.ensureCurrentPanel();
+
+      sinon.assert.calledOnce(Navigation.toPanel);
+      sinon.assert.calledWith(Navigation.toPanel, 'panel1');
+
+      // Let's be sure that navigation is finished and current panel is set
+      Navigation.toPanel.lastCall.returnValue.then(() => {
+        // Since we already have current panel, ensureCurrentPanel shouldn't
+        // call toPanel anymore.
+        Navigation.ensureCurrentPanel();
+
+        sinon.assert.calledOnce(Navigation.toPanel);
+      }).then(done, done);
+    });
+
+    test('does not navigate to default panel when transitioning is happening',
+    function() {
+      // We don't have any correct current panel, but we're in transitioning
+      // state currently that means that if nothing goes wrong we'll have
+      // initialized current panel soon and no need to force navigation to
+      // default panel.
+      Navigation.transitioning = true;
+      Navigation.ensureCurrentPanel();
+
+      sinon.assert.notCalled(Navigation.toPanel);
+
+      // When we're not in the transitioning state and current panel is not
+      // available then we should navigate to default panel;
+      Navigation.transitioning = false;
+      Navigation.ensureCurrentPanel();
```

Comment:
Mm, it's just for the sake of being more explicit in the test, will see how it will look with ```boolean ----> promise``` change for ```Navigation.transitioning```